uefi-test-runner/src/proto/media/known_disk.rs

```diff
@@ -138,6 +143,119 @@ fn test_create_file(directory: &mut Directory) {
     file.write(b"test output data").unwrap();
 }
 
+/// Tests raw disk I/O.
+fn test_raw_disk_io(handle: Handle, image: Handle, bt: &BootServices) {
+    info!("Testing raw disk I/O");
+
+    // Open the disk I/O protocol on the input handle
+    let disk_io = bt
+        .open_protocol::<DiskIo>(
+            OpenProtocolParams {
+                handle,
+                agent: image,
+                controller: None,
+            },
+            OpenProtocolAttributes::GetProtocol,
+        )
+        .expect("Failed to get disk I/O protocol");
+
+    // Allocate a temporary buffer to read into
+    const SIZE: usize = 512;
+    let buf = bt
+        .allocate_pool(MemoryType::LOADER_DATA, SIZE)
+        .expect("Failed to allocate temporary buffer");
+
+    // SAFETY: A valid buffer of `SIZE` bytes was allocated above
+    let slice = unsafe { core::slice::from_raw_parts_mut(buf, SIZE) };
+
+    // Read from the first sector of the disk into the buffer
+    disk_io
+        .read_disk(0, 0, slice)
```

Comment:
I think the first parameter (`media_id`) shouldn't be hardcoded here. If I'm understanding the spec correctly you have to also open the `BlockIO` protocol and get the [`media_id`](https://docs.rs/uefi/latest/uefi/proto/media/block/struct.BlockIOMedia.html#method.media_id) from that.